buildpack/telemetry/metrics.py

```diff
@@ -55,9 +55,37 @@
     ],
 }
 
+
+def get_apm_metrics_filter(env_var_name):
+    """
+    If we use empty string ("") in the filters that we use for statsd registry,
+    it accepts/denies every metric since we us type as `nameStartsWith`.
+    To prevent breaking the functionality because of this, we need to make sure
+    that we pass empty string to the registry filters only if it's intentional.
+    So, we strip the leading and trailing commas. Additionally we remove all
+     the white spaces to prevent any unintentional mistakes.
+    """
+    env_var_value = os.getenv(env_var_name)
+    if env_var_value is None:
+        return []
+    return env_var_value.replace(" ", "").strip(",").split(",")
```

Comment:
Not sure why we need the additional `strip` 🤔 

This should work right ?
```suggestion
    return env_var_value.replace(" ", "").split(",")
```

Reply:
If the there is a leading/trailing comma then split creates empty string in the resulting list which breaks our logic since the empty string means all metrics in this case. 
<img width="135" alt="image" src="https://user-images.githubusercontent.com/2044635/235951819-ded413bc-4f4d-41c7-81bd-3803e596f19b.png">

Reply:
Ah okay, got it. It's a bit of over-optimization though as to why we only care about the comma 😄

Reply:
We only care about comma because we split by comma :) We'll be waiting for comma separated values as input and the logic breaks if there is a leading/trailing comma in the input. So basically, I'm trying to prevent unintentional errors